File: src/analyze.py
from pathlib import Path

import pandas as pd


data_path = Path(__file__).resolve().parents[1] / "data" / "patients.csv"
patients = pd.read_csv(data_path)
site_counts = patients["site"].value_counts()

print("Patient Summary")
print(f"patient_count: {len(patients)}")
print(f"mean_age: {patients['age'].mean():.1f}")
print(f"minimum_age: {patients['age'].min()}")
print(f"maximum_age: {patients['age'].max()}")
print(f"site_DC: {site_counts.get('DC', 0)}")
print(f"site_MD: {site_counts.get('MD', 0)}")
print(f"site_VA: {site_counts.get('VA', 0)}")

File: scripts/analyze.R
patients <- read.csv(file.path("data", "patients.csv"))

summary_values <- dplyr::summarise(
  patients,
  patient_count = dplyr::n(),
  mean_age = mean(age),
  minimum_age = min(age),
  maximum_age = max(age)
)

cat("Patient Summary\n")
cat(sprintf("patient_count: %d\n", summary_values$patient_count))
cat(sprintf("mean_age: %.1f\n", summary_values$mean_age))
cat(sprintf("minimum_age: %d\n", summary_values$minimum_age))
cat(sprintf("maximum_age: %d\n", summary_values$maximum_age))
cat(sprintf("site_DC: %d\n", sum(patients$site == "DC")))
cat(sprintf("site_MD: %d\n", sum(patients$site == "MD")))
cat(sprintf("site_VA: %d\n", sum(patients$site == "VA")))
